addons/web/static/src/js/views/view_dialogs.js

```diff
@@ -461,15 +462,24 @@ var SelectCreateDialog = ViewDialog.extend({
                 classes: 'btn-primary o_select_button',
                 disabled: true,
                 close: true,
-                click: function () {
-                    var records = this.viewController.getSelectedRecords();
-                    var values = _.map(records, function (record) {
-                        return {
-                            id: record.res_id,
-                            display_name: record.data.display_name,
-                        };
-                    });
-                    this.on_selected(values);
+                click: async () => {
+                    if (this.viewController.isDomainSelected) {
+                        const state = this.viewController.model.get(this.viewController.handle, {raw: true});
+                        const ids = await this.viewController._domainToResIds(state.getDomain(), session.active_ids_limit)
+                        var values = _.map(ids, function (id) {
```

Comment:
use `const` everywhere, and `let` where you can't

Reply:
I'd create a public function on the ListController that basically does this
```js
        const state = this.model.get(this.handle, {raw: true});
        if (this.isDomainSelected) {
            resIds = await this._domainToResIds(state.getDomain(), session.active_ids_limit);
            displayNotif = (resIds.length === session.active_ids_limit);
        } else {
            resIds = this.model.localIdsToResIds(this.selectedRecords);
        }
        return resIds;
```
and would use it in Listcontroller and here

Reply:
Done :)